app/src/lib/git/reflog.ts

```diff
@@ -31,14 +34,26 @@ export async function getRecentBranches(
     return []
   }
 
-  const output = result.stdout
-  const lines = output.split('\n')
+  const lines = result.stdout.split('\n')
   const names = new Set<string>()
+  // exclude master from recent branches
+  const excludedNames = new Set<String>(['master'])
+
   for (const line of lines) {
     const result = regex.exec(line)
-    if (result && result.length === 2) {
-      const branchName = result[1]
-      names.add(branchName)
+    if (result && result.length === 4) {
+      const operationType = result[1]
+      const excludeBranchName = result[2]
+      const branchName = result[3]
+
+      if (operationType === 'renamed') {
```

Comment:
This condition would also be unnecessary with the suggested changes. We'd want to keep the addition of `excludeBranchName` to the set.